articles/file-handling-in-Cpp/index.md

```diff
@@ -0,0 +1,127 @@
+
+- **Portability**: With files, developers can transfer the data without worrying about data loss.
+
+In this article, we will walk through how to implement the File Handling function in C++, and we will cover the key working pointers of File handling such as:
+
+- Opening a File
+- Writing to a File
+- Reading from a File
+- Closing a file
+
+To perform File Handling functions, we need to add another library; fstream. The library holds all the functions related to File Handling. Here's how to include it.
+
+`#include<fstream>`
+
+Let's dive straight into learning.
+
+### Opening a File
+One of the most common actions in File Handling is opening a file. Let's see how to do it:
+
+```C
+ifstream fin;
+fin.open("index.txt")
+```
+
+That was simple as we just used the function **open** to open a file name "index.txt." but you as the reader may be wondering what is ` ifstream`
+
+**ifstream** is the type we use to declare an object. Very similar to the datatypes that we use in other functions. Such as: int.
+
+**fin** is the name of the variable. A developer can name it anything they want.
+
+### Writing in a File
+Writing to a file is easier than even opening a file. Instead of using **cout**, we just need to use the name of the variable that we recently selected. Here's how to do it:
+
+```C
+fin<<"Section.io is a great platform to learn about new technologies."<<endl;
+```
+
+### Reading from a File
+We first need to declare a variable similar to the type of data in our file. Then run a while loop until the very end of the file and cout all the data. It sounds a bit complex, but it isn't. Here's how to do it:
+
+```C
+string data;
+while(fin >> data) {
+cout<<data;
+}
+```
+
+### Close a File
+Closing a file is the easiest of all the actions we've done so far. Here's how to do it:
+
+```C
+fin.close();
+```
+
+It's just that simple.
+
+So far, we have covered everything that a beginner needs to know about file handling, or you could say everything that you will ever use. Now let's write a program to implement what we have just covered to get a clearer picture.
+
+```C
+#include<iostream>
+#include<fstream>
+
+using namespace std;
+
+int main() {
+ifstream file;
+string file_data;
+
+file.open("index.md"); //Opening a File
+fin<<"Section.io is a great platform to learn about new technologies."; //Writing in a File
+
+while(file >> file_data) { //Redaing from a File
+cout<<data;
+}
+file.close(); //Closing a File
+
+return 0;
+}
+```
+
+In the above program , we implemented everything that we have covered so far. Let's break it down and see what's going on in the above program.
+
+- **Opening a File**
+
+To open a file, we used a function ```file.open("index.md")``` that open a file named **index.md**. That how's simple the command is to open a file in File Handling.
```

Comment:
that open a file named -> that opens a file named

Reply:
"That how's simple the command is to open a file in File Handling"

*you already mentioned this above*